File: 프로그래머스/레벨1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const stack = new Array(ingredient.length).fill(0);
    let top = -1;
    let result = 0;
    
    for (const item of ingredient) {
        stack[++top] = item;
        
        if (top >= 3) {
            if (stack[top] === 1 && stack[top-1] === 3 && stack[top-2] === 2 && stack[top-3] === 1) {
                top -= 4;
                result++;
            }
        }
    }
    
    return result;
}

File: 프로그래머스/레벨1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    vector<int>stack(ingredient.size(),0);
    int top = -1;
    int result = 0;
    
    for (int item : ingredient) {
        stack[++top] = item;

        if (top >= 3) {
            if (stack[top] == 1 && stack[top-1] == 3 && stack[top-2] == 2 && stack[top-3] == 1) {
                top -= 4;
                result++;
            }
        }
    }

    return result;    
}